File: src/Entity/Formulaire.php
<?php

namespace App\Entity;

use App\Repository\FormulaireRepository;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;
/**
 * @ORM\Entity(repositoryClass=FormulaireRepository::class)
 */
class Formulaire
{


    /**
     * @ORM\Id
     * @Assert\Positive(message="ce champs doit etre positive")
     * @Assert\NotBlank(message="ce champs est obligatoire")
     * @ORM\Column(type="integer")
     */
    private $ref;

    /**
     * @Assert\Positive(message="ce champs doit etre positive")
     * @Assert\Length(min=8, minMessage="entrer un nombre de taille egale a 8")
     * @Assert\NotBlank(message="ce champs est obligatoire")
     * @ORM\Column(type="integer")
     *
     */
    private $cin;

    /**
     * @Assert\NotBlank(message="ce champs est obligatoire")
     * @ORM\Column(type="string", length=255)
     */
    private $quest1;

    /**
     * @Assert\NotBlank(message="ce champs est obligatoire")
     * @ORM\Column(type="string", length=255)
     */
    private $quest2;

    /**
     * @Assert\NotBlank(message="ce champs est obligatoire")
     * @ORM\Column(type="string", length=255)
     */
    private $quest3;

    /**
     * @Assert\NotBlank(message="ce champs est obligatoire")
     * @ORM\Column(type="string", length=255)
     */
    private $quest4;

    /**
     * @Assert\NotBlank(message="ce champs est obligatoire")
     * @ORM\Column(type="string", length=255)
     */
    private $quest5;

    /**
     * @Assert\NotBlank(message="ce champs est obligatoire")
     * @ORM\Column(type="string", length=255)
     */
    private $quest6;

    /**
     * @Assert\NotBlank(message="ce champs est obligatoire")
     * @ORM\Column(type="string", length=255)
     */
    private $type;

    /**
     * @ORM\ManyToOne(targetEntity=FormCateg::class, inversedBy="formulaire")
     * @ORM\JoinColumn(nullable=false ,name="FormCateg", referencedColumnName="id_cat")
     */
    private $FormCateg;







    public function getRef(): ?int
    {
        return $this->ref;
    }

    public function setRef(int $ref): self
    {
        $this->ref = $ref;

        return $this;
    }

    public function getCin(): ?int
    {
        return $this->cin;
    }

    public function setCin(int $cin): self
    {
        $this->cin = $cin;

        return $this;
    }

    public function getQuest1(): ?string
    {
        return $this->quest1;
    }

    public function setQuest1(string $quest1): self
    {
        $this->quest1 = $quest1;

        return $this;
    }

    public function getQuest2(): ?string
    {
        return $this->quest2;
    }

    public function setQuest2(string $quest2): self
    {
        $this->quest2 = $quest2;

        return $this;
    }

    public function getQuest3(): ?string
    {
        return $this->quest3;
    }

    public function setQuest3(string $quest3): self
    {
        $this->quest3 = $quest3;

        return $this;
    }

    public function getQuest4(): ?string
    {
        return $this->quest4;
    }

    public function setQuest4(string $quest4): self
    {
        $this->quest4 = $quest4;

        return $this;
    }

    public function getQuest5(): ?string
    {
        return $this->quest5;
    }

    public function setQuest5(string $quest5): self
    {
        $this->quest5 = $quest5;

        return $this;
    }

    public function getQuest6(): ?string
    {
        return $this->quest6;
    }

    public function setQuest6(string $quest6): self
    {
        $this->quest6 = $quest6;

        return $this;
    }

    public function getType(): ?string
    {
        return $this->type;
    }

    public function setType(string $type): self
    {
        $this->type = $type;

        return $this;
    }

    public function getFormCateg(): ?FormCateg
    {
        return $this->FormCateg;
    }

    public function setFormCateg(?FormCateg $FormCateg): self
    {
        $this->FormCateg = $FormCateg;

        return $this;
    }



}

File: sprint java/Entities/Formulaire.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package Entities;

import java.util.Comparator;

/**
 *
 * @author amine
 */
public class Formulaire {
    private  int ref,cin,FormCateg;
    private String quest1,quest2,quest3,quest4,quest5,quest6,type; 

  

    public Formulaire( int cin, String quest1, String quest2, String quest3, String quest4, String quest5, String quest6, String type,int FormCateg) {
        
        this.cin = cin;
        this.quest1 = quest1;
        this.quest2 = quest2;
        this.quest3 = quest3;
        this.quest4 = quest4;
        this.quest5 = quest5;
        this.quest6 = quest6;
        this.type = type;
        this.FormCateg = FormCateg;

    }

    public int getFormCateg() {
        return FormCateg;
    }

    public void setFormCateg(int FormCateg) {
        this.FormCateg = FormCateg;
    }

    public Formulaire() {
        
    }

    public void setRef(int ref) {
        this.ref = ref;
    }

    public int getRef() {
        return ref;
    }

    public int getCin() {
        return cin;
    }

    public void setCin(int cin) {
        this.cin = cin;
    }

    public String getQuest1() {
        return quest1;
    }

    public void setQuest1(String quest1) {
        this.quest1 = quest1;
    }

    public String getQuest2() {
        return quest2;
    }

    public void setQuest2(String quest2) {
        this.quest2 = quest2;
    }

    public String getQuest3() {
        return quest3;
    }

    public void setQuest3(String quest3) {
        this.quest3 = quest3;
    }

    public String getQuest4() {
        return quest4;
    }

    public void setQuest4(String quest4) {
        this.quest4 = quest4;
    }

    public String getQuest5() {
        return quest5;
    }

    public void setQuest5(String quest5) {
        this.quest5 = quest5;
    }

    public String getQuest6() {
        return quest6;
    }

    public void setQuest6(String quest6) {
        this.quest6 = quest6;
    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    @Override
    public String toString() {
        return "Formulaire{" + "ref=" + ref + ", cin=" + cin + ", FormCateg=" + FormCateg + ", quest1=" + quest1 + ", quest2=" + quest2 + ", quest3=" + quest3 + ", quest4=" + quest4 + ", quest5=" + quest5 + ", quest6=" + quest6 + ", type=" + type + '}';
    }

  
         public static Comparator <Formulaire> ComparatorREF =(Formulaire o1, Formulaire o2) -> o1.getRef()- o2.getRef();
    
}
